### PR TITLE
Make use of the #derive(JsonSchema) addition to progenitor

### PR DESCRIPTION
**Pull request by smklein**
Relies on https://github.com/oxidecomputer/progenitor/pull/21

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -57,8 +57,10 @@ panic = "abort"
 #
 # Local client generation during development.
 #
-#[patch."https://github.com/oxidecomputer/progenitor"]
-#progenitor = { path = "../progenitor/progenitor" }
+[patch."https://github.com/oxidecomputer/progenitor"]
+progenitor = { path = "../progenitor/progenitor" }
+[patch."https://github.com/oxidecomputer/crucible"]
+crucible-agent-client = { path = "../crucible/agent-client" }
```
Comment (smklein) [inline Cargo.toml:63]: These obviously need to be removed before merging.

File: oximeter/producer/src/lib.rs [hunk 2]
```diff
@@ -180,7 +180,8 @@ pub async fn register(
     client
         .cpapi_producers_post(&server_info.into())
         .await
-        .map_err(|msg| Error::RegistrationError(msg.to_string()))
+        .map_err(|msg| Error::RegistrationError(msg.to_string()))?;
+    Ok(())
```
Comment (ahl) [inline oximeter/producer/src/lib.rs:184]: yeah, this is an unrelated progenitor update that we'll catch on the next dependabot sync

**Comment by smklein** [comment]
I went ahead and merged this into https://github.com/oxidecomputer/omicron/pull/511 , closing.

**Comment by ahl** [comment]
@smklein did this make it in? https://github.com/oxidecomputer/omicron/pull/580/commits/1beee373284675cfce9204bf11ee697d404e2768#diff-d38884f7391ae50f795707dbb281656e8afe38ea3ecac15a6ee80d52724017c0

**Comment by smklein** [comment]
Admittedly, no, it did not - I think the `JsonSchema` I actually need is the one for the Crucible Agent Client, which I believe already merged in Crucible: https://github.com/oxidecomputer/crucible/blob/main/agent-client/src/lib.rs#L7

**Comment by ahl** [comment]
Nice
